pkg/validation/common.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"github.com/go-openapi/spec"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+	"reflect"
+)
+
+type FieldMeta struct {
+	*field.Path
+	Type string // only if it is builtin
+}
+
+type OperationType int
+
+const (
+	CREATE OperationType = iota
```

Comment:
Maybe POST/PUT

Not sure what exactly it means to do field validation on a DELETE?